### PR TITLE
Clean up filename and required ignition version templates

### LINKED ISSUE #37: Align required Ignition/platform version with sdk_version extra property on generation
I don't think there's any harm in defaulting these two properties to align; anyone who knows how to untangle them will probably already understand the ramifications of doing so. But we can default to a safer/saner alignment.

### LINKED ISSUE #38: Update generated skeleton fileName property description to clarify the file extension
image

Whichever way is correct (I believe, **with** file extension) we should make sure the generated files are correct.

### PR DESCRIPTION
**Pull request by paul-griffith**
Fixes #37 
Fixes #38

### CODE REVIEW

**Comment by brianeray** [comment]
Hey @paul-griffith. Is this PR still alive? I can  review if needed.

And get a clean build; looks like the logs are gone.

**Comment by paul-griffith** [comment]
Haven't touched it in a long time, obviously, but I think the changes are still valid.

File: generator/README.md [hunk 1]
```diff
@@ -3,9 +3,9 @@
...
 Gradle build tool.
 
-1. [Ignition Module Generator Core](https://github.com/inductiveautomation/ignition-module-tools/tree/master/generator/generator-core#readme) library for creating Ignition module skeletons.  Written in Kotlin, usable in Kotlin, Java and other JVM languages via jar-based dependency available as Maven artifacts.
+1. [Ignition Module Generator Core](generator-core/README.md) library for creating Ignition module skeletons.  Written in Kotlin, usable in Kotlin, Java and other JVM languages via jar-based dependency available as Maven artifacts.
 
-2. [Module Generator CLI](https://github.com/inductiveautomation/ignition-module-tools/tree/master/generator/generator-cli#readme) command line application which uses the generator library to provide a _command line interface_ (CLI) that can be used to create modules from your operating system's terminal/console.
+2. [Module Generator CLI](/generator-cli/README.md) command line application which uses the generator library to provide a _command line interface_ (CLI) that can be used to create modules from your operating system's terminal/console.
```
Comment (brianeray) [inline generator/README.md:8]: Pretty sure leading slash anchors the link to the repo root.

```suggestion
2. [Module Generator CLI](generator-cli/README.md) command line application which uses the generator library to provide a _command line interface_ (CLI) that can be used to create modules from your operating system's terminal/console.
```

File: generator/generator-core/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ A library for generating boilerplate folder/file structures for gradle-based Ign
 
 We've had requests to make it easier to get started with module development using Gradle. So when we started writing
 functional tests for
-the [Gradle Module Plugin](https://github.com/inductiveautomation/ignition-module-tools/tree/master/gradle-module-plugin#readme)
+the [Gradle Module Plugin](https://github.com/inductiveautomation/ignition-module-tools/tree/master/gradle-module-plugin)
```
Comment (brianeray) [inline generator/generator-core/README.md:9]: I think you can do this for more portability.

```suggestion
the [Gradle Module Plugin](../../gradle-module-plugin)
```

**Review by brianeray — COMMENTED** [review]
I'm going to see if I can manually kick off the GH workflow again to find out why it failed the last time.

**Review by brianeray — APPROVED** [review]

**Comment by brianeray** [comment]
I don't have privileges to override the checks failure and merge, which is what I would do since these aren't code changes. I might need to bug the `inductiveautomation` org owners.

**Comment by KathyApplebaum** [comment]
@brianeray Just stumbled across this. Can you explain to me in simple terms why it's okay to merge with a failed build? Thanks!

**Comment by brianeray** [comment]
No. Looking at this again, especially this,

> what I would do since these aren't code changes

`README.md`'s, OK, perhaps even the changes to the templates. But there's a logic change in `ModuleGeneratorContext`. What I wrote makes no sense.

This thing should pass a PR build first.

**Comment by brianeray** [comment]
Wait a second. I'm crazy, but not that crazy. My comment was on March 16. As of March 16, the PR _was_ simply `README.md` changes:

image

The logic fix came two months later.

So now for sure, it should have to pass a PR build.

**Comment by brianeray** [comment]
Trying to get to the PR build failure yields a dead end:

image

The PR workflow has succeeded for other PRs since then, however. This is weird.

**Comment by brianeray** [comment]
OK I finally tricked it into [rebuilding](https://github.com/inductiveautomation/ignition-module-tools/actions/runs/10222310704) and giving us good output. I've downloaded the failed Gradle log and finally have something to go on.

Stay tuned.

**Comment by brianeray** [comment]
For internal tracking: IGN-10603.

**Comment by brianeray** [comment]
@paul-griffith I think I figured it out. It's coming from the current PR. Try this.

```text
> git diff --% --ignore-all-space --color-moved --color-moved-ws=ignore-all-space --
diff --git before:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.groovy after:generator/generator-core/src/main/resources/templates/config/modlPluginC
onfig.groovy
index d2ca25a..22eb18e 100644
--- before:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.groovy
+++ after:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.groovy
@@ -22,7 +22,7 @@ ignitionModule {
      * the course of a major Ignition (7.9, 8.0, etc) version, except for when the Ignition Platform adds/changes APIs
      * used by the module.
      */
-    requiredIgnitionVersion = ext.sdk_version
+    requiredIgnitionVersion = rootProject.ext.sdk_version
     /*
      *  This is a map of String: String, where the 'key' represents the fully qualified path to the project
      *  (using gradle path syntax), and the value is the shorthand Scope string.
diff --git before:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.kts after:generator/generator-core/src/main/resources/templates/config/modlPluginConf
ig.kts
index 176d0db..83122e5 100644
--- before:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.kts
+++ after:generator/generator-core/src/main/resources/templates/config/modlPluginConfig.kts
@@ -25,7 +25,7 @@ ignitionModule {
      * the course of a major Ignition (7.9, 8.0, etc) version, except for when the Ignition Platform adds/changes APIs
      * used by the module.
      */
-    requiredIgnitionVersion.set(sdk_version)
+    requiredIgnitionVersion.set(rootProject.ext.sdk_version)
     /*
      *  This is a map of String: String, where the 'key' represents the fully qualified path to the project
      *  (using gradle path syntax), and the value is the shorthand Scope string.
```

I'm having problems getting unit tests to pass even after that patch due to Gradle-Windows fighting, but am pretty sure this squashes the thing that is tanking most of the unit tests.

[1]

```text
io.ia.sdk.gradle.modl.IgnitionModulePluginFunctionalTest > gateway scoped unsigned module passes config and builds with signing credentials STANDARD_OUT
    /tmp/junit5484288864958444835/gwScopedUnsignedConfigAndBuild

    FAILURE: 
    Build failed with an exception.


    * Where:

    Build file '/tmp/junit5484288864958444835/gwScopedUnsignedConfigAndBuild/great-tests/build.gradle' line: 46


    * What went wrong:

    A problem occurred evaluating root project 'great-tests'.

    > 
    Cannot get property 'sdk_version' on extra properties extension as it does not exist

 ... and the same thing in dozens more test failures ...
```

**Comment by brianeray** [comment]
> I'm having problems getting unit tests to pass even after that patch due to Gradle-Windows fighting

OK this is better; I'm not going to run the whole suite locally but it gives confidence in the `sdk_version` patch.

```text
> .\gradlew.bat --% :gradle-module-plugin:functionalTest --tests=AssembleModuleStructureTest."single file doc is collected"
Type-safe project accessors is an incubating feature.

 ...

BUILD SUCCESSFUL in 1m 20s
12 actionable tasks: 3 executed, 9 up-to-date
```

**Comment by brianeray** [comment]
Replaced by #58--just to make it easier for me to grind away at the unit test failures and take this off of Paul's plate.

**Pull request closed without merging**